src/sentry/web/api.py

```diff
@@ -227,26 +227,13 @@ def _dispatch(self, request, helper, project_id=None, origin=None, *args, **kwar
             project.organization = Organization.objects.get_from_cache(
                 id=project.organization_id)
 
-            if auth.version != '2.0':
-                if not auth.secret_key:
-                    # If we're missing a secret_key, check if we are allowed
-                    # to do a CORS request.
-
-                    # If we're missing an Origin/Referrer header entirely,
-                    # we only want to support this on GET requests. By allowing
-                    # un-authenticated CORS checks for POST, we basially
-                    # are obsoleting our need for a secret key entirely.
-                    if origin is None and request.method != 'GET':
-                        raise APIForbidden(
-                            'Missing required attribute in authentication header: sentry_secret'
-                        )
-
-                    if not is_valid_origin(origin, project):
-                        if project:
-                            tsdb.incr(
-                                tsdb.models.project_total_received_cors, project.id)
-                        raise APIForbidden(
-                            'Missing required Origin or Referer header')
+            if auth.version != '2.0' and \
```

Comment:
according to codecov this isnt tested -- could we get something that exercises this path of code (or tell me codecov is wrong)

probably also worth rebasing if its been a while